### PR TITLE
fix image size

### PR DESCRIPTION
**Pull request by fahedouch**
the actual image size is equal to the ChainID size.

this PR use the Walk API and Usage on each parent of  ChainID to calculate the total usage of unpacked image layers

related issue   https://github.com/containerd/nerdctl/issues/304 

Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### CODE REVIEW

File: cmd/nerdctl/images.go [hunk 1]
```diff
@@ -143,19 +146,40 @@ func imagesBashComplete(clicontext *cli.Context) {
...
+	keys.add(ctx, s, &usage, keys[chainID])
 	return usage.Size, nil
 }
+
+// recursive function to calculate total usage of ChainID's parent
+func (keys snapshotKey) add(ctx context.Context, s snapshots.Snapshotter, usage *snapshots.Usage, key string) {
+	if key == "" {
+		return
+	}
+	u, _ := s.Usage(ctx, key)
```
Comment (ktock) [inline cmd/nerdctl/images.go:182]: Error shouldn't be ignored

File: cmd/nerdctl/images.go [hunk 2]
```diff
@@ -143,19 +146,40 @@ func imagesBashComplete(clicontext *cli.Context) {
...
+	usage.Add(u)
+	keys.add(ctx, s, usage, keys[key])
+}
```
Comment (ktock) [inline cmd/nerdctl/images.go:185]: This (`snapshotKey.add`) should be put immediately following the type definition of `snapshotKey` (`type snapshotKey map[string]string`) .

**Review by ktock — COMMENTED** [review]
overall LGTM but minor nits

**Comment by fahedouch** [comment]
@ktock done

File: cmd/nerdctl/images.go [hunk 3]
```diff
@@ -143,19 +146,47 @@ func imagesBashComplete(clicontext *cli.Context) {
...
 	}
 
+	keys := make(snapshotKey, 0)
+	walkFn := func(ctx context.Context, info snapshots.Info) error {
+		keys[info.Name] = info.Parent
+		return nil
+	}
+	if err := s.Walk(ctx, walkFn); err != nil {
+		return 0, err
+	}
```
Comment (ktock) [inline cmd/nerdctl/images.go:185]: Why does it scan all snapshots? Can we walk snapshots only in the chain? I think we can do this by following parents  from the topmost snapshot using `Stat` API.

Comment (fahedouch) [reply cmd/nerdctl/images.go]: @ktock thank you,  This is what I was looking for ==> `Stat` API . I made modification.

**Comment by AkihiroSuda** [comment]
CI wasn't triggered?

File: cmd/nerdctl/images.go [hunk 4]
```diff
@@ -143,19 +146,51 @@ func imagesBashComplete(clicontext *cli.Context) {
 	bashCompleteImageNames(clicontext)
 }
 
-func unpackedImageSize(ctx context.Context, clicontext *cli.Context, client *containerd.Client, i images.Image) (int64, error) {
+type snapshotKey string
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:149]: Why type?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**